src/rpc/rawtransaction.cpp

```diff
@@ -882,6 +882,11 @@ UniValue SignTransaction(CMutableTransaction& mtx, const UniValue& prevTxsUnival
         }
         sigdata = CombineSignatures(prevPubKey, TransactionSignatureChecker(&txConst, i, amount), sigdata, DataFromTransaction(mtx, i));
 
+        // amount required for valid segwit signature
+        if (amount == MAX_MONEY && !sigdata.scriptWitness.IsNull()) {
```

Comment:
Please do ignore this if comparing against `MAX_MONEY` is valid, but if it isn't and you will be using an amount outside of the range `[0,MAX_MONEY]` then the boolean expression needs to be addressed. I suspect something along these lines would work:
```c++
if (!MoneyRange(amount) && !sigdata.scriptWitness.IsNull()) {
```